### PR TITLE
TF Lite unit (kernel) tests CMake build fix

### PR DESCRIPTION
**Pull request by madaosik**
The recent changes in the TF Lite kernel tests architecture were reflected in Bazel build system **but not in CMake**, leading to runtime failure when built using this platform.

This change **updates the TF Lite kernel tests CMake build configuration** and adds a minor check for easier future debugging.

### CODE REVIEW

**Comment by madaosik** [comment]
@googlebot I signed it!

File: tensorflow/lite/kernels/test_delegate_providers.cc [hunk 1]
```diff
@@ -29,6 +29,8 @@ constexpr char KernelTestDelegateProviders::kUseSimpleAllocator[];
 
 KernelTestDelegateProviders::KernelTestDelegateProviders()
     : delegate_list_util_(&params_) {
+  TFLITE_TOOLS_CHECK(delegate_list_util_.GetCount() > 0)
```
Comment (multiverse-tf) [inline tensorflow/lite/kernels/test_delegate_providers.cc:32]: It's possible that the test could run without any delegate providers. So, this check could fail?

File: tensorflow/lite/kernels/CMakeLists.txt [hunk 2]
```diff
@@ -41,18 +41,37 @@ build_flatbuffers(
...
+set(DELEGATE_PROVIDERS_SUPP
+  ${TFLITE_SOURCE_DIR}/delegates/external/external_delegate.cc
+  ${TFLITE_SOURCE_DIR}/nnapi/sl/SupportLibrary.cc
+  ${TFLITE_SOURCE_DIR}/tools/delegates/delegate_provider.cc
+  ${TFLITE_SOURCE_DIR}/tools/evaluation/utils.cc
+)
+
+set(DELEGATE_PROVIDERS
+  ${DELEGATE_PROVIDERS_SUPP}
+  ${TFLITE_SOURCE_DIR}/tools/delegates/coreml_delegate_provider.cc
```
Comment (multiverse-tf) [inline tensorflow/lite/kernels/CMakeLists.txt:53]: I think some delegate providers are only compilable on certain platforms, e.g., the CoreML delegate might be only compilable for iOS, and when using Bazel, we only add this dependency for iOS. So, I think we should not blindly add all providers here.

File: tensorflow/lite/tools/delegates/delegate_provider.h [hunk 3]
```diff
@@ -157,6 +157,9 @@ class ProvidedDelegateList {
     return CreateAllRankedDelegates(*params_);
   }
 
+  // Returns the count of referenced delegates in providers_ variable
+  const int GetCount();
```
Comment (multiverse-tf) [inline tensorflow/lite/tools/delegates/delegate_provider.h:161]: Might simply inline the implementation here? Btw, if the change to tensorflow/lite/kernels/test_delegate_providers.cc were dropped, this function would be no longer needed?

File: tensorflow/lite/kernels/CMakeLists.txt (hunk 2 above)
Comment (terryheo) [reply tensorflow/lite/kernels/CMakeLists.txt:53]: right. Also NNAPI delegate should available only for Android

File: tensorflow/lite/kernels/test_delegate_providers.cc (hunk 1 above)
Comment (madaosik) [reply tensorflow/lite/kernels/test_delegate_providers.cc:32]: It is possible to run the test without any delegate provider. It is compulsory, though, to have at least **DefaultExecutionProvider** instance available acting as a "dummy delegate" providing the common parameters and flags. Without it the test fails during runtime based on the missing "help" parameter (as described [here](https://github.com/tensorflow/tensorflow/pull/52110#discussion_r718332050)). I admit, nevertheless, that the error message was rather misleading, so I updated it in the latest commit.

File: tensorflow/lite/kernels/CMakeLists.txt (hunk 2 above)
Comment (madaosik) [reply tensorflow/lite/kernels/CMakeLists.txt]: Good point. Let me put it another way. The ([tensorflow/lite/CMakeLists.txt](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/CMakeLists.txt)) provides various _options_ using which the NNAPI, GPU and XNNPACK delegates can be enabled/disabled. As the user can toggle these _options_ even after the _configure_ phase is finished, it is required to have them in the _DELEGATE_PROVIDERS_ list. Conversely and admittedly, the delegates not mentioned in the root CMakeLists (Hexagon, External, GPU) are irrelevant and therefore were removed in the latest commit.

File: tensorflow/lite/tools/delegates/delegate_provider.h (hunk 3 above)
Comment (madaosik) [reply tensorflow/lite/tools/delegates/delegate_provider.h]: Implementation has been inlined in the latest commit. The answer is yes. However I would suggest keeping it for the reasons discussed above.

**Review by multiverse-tf — COMMENTED** [review]
This overall looks good to me. But I'd like to wait for @terryheo's approval as he is the owner of TFLite CMake support.

**Comment by madaosik** [comment]
Merged already as part of https://github.com/tensorflow/tensorflow/pull/52110.

**Pull request closed without merging**